Test1/src/leetcode/LeetCode34.java

```diff
@@ -0,0 +1,57 @@
+package leetcode;
+
+/**
+ * https://leetcode.com/problems/find-first-and-last-position-of-element-in-sorted-array/ 
+ * @author minibig
+ */
+public class LeetCode34 {
+	
+    public int[] searchRange(int[] nums, int target) {
+        int[] answer = new int[] {-1,-1};
+        if(nums.length == 0){
+            return answer;
+        }
+        
+        int l = 0;
+        int r = nums.length - 1;
+        
+        int midIdx = (l + r) / 2;
```

Comment:
Not pretty sure why you put this outside.

I think you can directly calculate midIdx in each iteration in the while loop.